ams/parser.js

```diff
@@ -65,59 +65,50 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     if (param.doi) {
       result.unitid = param.doi.split('/')[1];
     }
-  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && param.tab_body == 'pdf') {
+  } else if ((match = /^\/(view|downloadpdf)\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.(xml|jpg)$/i.exec(path)) !== null) {
     // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=pdf
-    result.rtype = 'ARTICLE';
-    result.mime  = 'PDF';
-    result.title_id = match[1];
-    result.vol = match[2];
-    result.issue = match[3];
-    result.unitid = match[4];
-  } else if ((match = /^\/downloadpdf\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null) {
     // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=pdf
     // https://journals.ametsoc.org/downloadpdf/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml
-    result.rtype = 'ARTICLE';
-    result.mime  = 'PDF';
-    result.title_id = match[1];
-    result.vol = match[2];
-    result.issue = match[3];
-    result.unitid = match[4];
-
-  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/([a-z0-9-.]+)\.xml$/i.exec(path)) !== null && (param.tab_body == 'fulltext-display' || param.tab_body == null)) {
     // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=fulltext-display
-    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0009.1.xml
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0009.1.xml    
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/full-amsmonographs-d-18-0005.1-f2.jpg
+    // https://journals.ametsoc.org/view/journals/amsm/59/1/amsmonographs-d-18-0005.1.xml?tab_body=abstract-display
+
     result.rtype = 'ARTICLE';
     result.mime  = 'HTML';
-    result.title_id = match[1];
-    result.vol = match[2];
-    result.issue = match[3];
-    result.unitid = match[4];
 
-  } else if ((match = /^\/view\/journals\/([a-z]+)\/([0-9]+)\/([0-9]+)\/[a-z]+-([a-z0-9-.]+)\.jpg$/i.exec(path)) !== null) {
-    // https://journals.ametsoc.org/view/journals/amsm/59/1/full-amsmonographs-d-18-0005.1-f2.jpg
-    result.rtype = 'IMAGE';
-    result.mime  = 'JPEG';
-    result.title_id = match[1];
-    result.vol = match[2];
-    result.issue = match[3];
-    result.unitid = match[4];
+    if (match[1] === 'downloadpdf' || param.tab_body === 'pdf') {
+      result.mime  = 'PDF';
+    }
```

Comment:
After a further analysis of the platform, we can not count the access via tab_body=pdf as a PDF consultation because the viewer is in an HTML page, so it must be considered as

```js
if (match[1] === 'downloadpdf') {
  result.mime = 'PDF';
}

if (param.tab_body === 'PDF') {
  result.mime = HTML
}
```

Reply:
I firmly disagree. The user is viewing the PDF version of the document not an HTML version. There is a CLEAR difference between viewing an HTML version which often doesn't contain the same content, charts, graphs and mathematical formulas in particular and the PDF which is being displayed via an HTML "viewer". This would be like saying that if you're using an HTML5 viewer for video in MP4 format that shouldn't count as viewing the VIDEO